New project layout, Pod 1.0+Carthage+SPM [WIP]

fulldecent opened the pull request:
## Wording:

 * pod-template -- this project
 * template -- a model that can have things changed
 * vendored -- a chosen template with variable substitutions and other updates applied and ready for making an iOS framework

## pod-template progress:

- [x] Make vendoring work
- [x] Set up testing with rake/rspec to validate the vendoring process (fixes #79)
- [x] Set up Travis CI
- [x] Update variable substitution process, remove unused code

## baseline template progress: (new feature)

- [x] Move all files common to Swift and Objective-C into a "baseline" template
- [x] Update to latest gitignore
- [x] Supports Swift Package Manager format (fixes #157)

## swift template progress:

- [x] Create a hello world example (this is a change), it actually does something
- [x] Remove FDChessboardView and my name from project, use variable substitution
- [x] Remove XCode warnings (fixes #131)

## objective-c template progress: [HELP REQUESTED]

- [ ] Translate the current Swift template into Objective-C

## other blockers:

**PLEASE HELP REVIEW AND ADD BLOCKERS HERE**

- [ ] Confirm each set of options works [HELP REQUESTED]

orta commented:
So, this removes a bunch of features that are not optional, like all the questions, swift / objc support, library choices, optional example applications all of which are blockers on this PR.

Other notes, 
- I don't want the current README for their library changing. 
- I also think at first we shouldn't ship a `package.swift` file, there's no point in shipping one that won't work by default simply because AFNetworking does - it'll totally throw off anyone trying to know if something supports SPM or not when there's thousands of libraries that have an SPM package but it's not set up or configured right 

It's got a while to go, but I'm excited to see progress 👍

fulldecent commented:
I fixed the README and removed explicit SPM support.

Also added some VERY BASIC travis CI and testing for this repo. I've never used rspec (or ruby) before but this should be good minimally viable and others can improve :-). There is one error related to CocoaPods but I think we can still commit with that error.

fulldecent commented:
Update issue and added work list. Requesting review and help from anyone interested.

fulldecent commented:
Right now everything should be working except for the objective-c template.

Please anyone assist with testing this and commenting on anything that it broken.

orta commented:
This definitely has a way to go yet, tests crash on my computer, and even the perfect example ends up un-usable. I want to do some work on the CocoaPods-App 1.0 for the next few days, then I can come and give this polish to make it shippable as a PR to you 👍 

There's some nice ideas in here

![screen shot 2016-05-10 at 2 56 01 pm](https://cloud.githubusercontent.com/assets/49038/15158773/5b83cd54-16bf-11e6-8f68-2585cf2430db.png)

fulldecent commented:
@orta Thanks for your help!

fulldecent commented:
This is what I am using to successfully execute the tests:
 
    git clone git@github.com:fulldecent/pod-template.git
    cd pod-template/
    bundle install --jobs=3 --retry=3
    rake

segiddins commented:
Looks like the `.travis.yml` needs to specify a ruby version in it

fulldecent commented:
Testing is killing me -- time for bed :-(

fulldecent commented:
Now testing is somewhat working

fulldecent commented:
To all: please advise and help. What needs to be done here and how can we do it?

orta commented on .gitignore:
```diff
@@ -26,8 +26,12 @@ Carthage
 # Pods/
+
+Gemfile.lock
+play
```
Where does this come from?

fulldecent replied:
These updates use Rake (installed by Gemfile) for testing in Travis CI.

orta commented on configurator/ProjectManipulator.rb:
```diff
@@ -89,11 +81,11 @@ def project_folder
       # rename xcproject
-      File.rename(project_folder + "/PROJECT.xcodeproj", project_folder + "/" +  @configurator.pod_name + ".xcodeproj")
+#      File.rename(project_folder + "/PROJECT.xcodeproj", project_folder + "/" +  @configurator.pod_name + ".xcodeproj")
 
```
Do these want to stay commented?

fulldecent replied:
Thank you. No, this is deleted in a38d43973dda8e7a1a868989825c43086171581e

orta commented on templates/swift/Example/Podfile:
```diff
@@ -1,11 +1,11 @@
+source 'https://github.com/CocoaPods/Specs.git'
```
Why did you add this? ( It's not needed )

fulldecent replied:
Removed in ed0103ec67adbc4bbbb7ccc18ff8b1b3a8848fd3

orta commented on templates/swift/Example/Source/AppDelegate.swift:
```diff
@@ -1,9 +1,9 @@
-//  Created by PROJECT_OWNER on TODAYS_DATE.
-//  Copyright (c) TODAYS_YEAR PROJECT_OWNER. All rights reserved.
+//  Created by Full Decent on 4/25/16.
+//  Copyright © 2016 William Entriken. All rights reserved.
```
^

fulldecent replied:
Thank you, fixed at 1319e9ff8e164a12786f1e60d379d18a0be5923d

orta commented on templates/swift/Example/Source/ViewController.swift:
```diff
@@ -0,0 +1,18 @@
+
+class ViewController: UIViewController {
+    @IBOutlet var chessboard: PROJECT!
+
```
This isn't a generic, set up template example template yet ^

fulldecent replied:
Thank you, fixed with cf7656fe299594d4e1dae6d2a91d1dbce43414b7

orta commented on templates/swift/README.md:
```diff
@@ -0,0 +1,108 @@
+See Also
+===========
+
+See also Kibitz for Mac which is making a comeback https://github.com/fulldecent/kibitz
```
This is not the README for a new project ^

fulldecent replied:
Fixed in cf7656fe299594d4e1dae6d2a91d1dbce43414b7

orta commented:
Given that this PRs into master, it cannot be merged until it's 100% production ready - as everyone instantly starts using it, there is no deploy process. So it'll need to have the exact same feature set from before. 

You're welcome to PR to a `develop` branch ( I've just pushed master to it right now ) and then I can help out on some of these, but it's still got some work to do 👍

fulldecent commented:
@orta Thank you for ALL your help. I am closing this PR and have opened another one against the Develop branch per your note.

https://github.com/CocoaPods/pod-template/pull/179
